mautic/mautic
# Fix salesforce sync companies even are disabled

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR prevent  error of Salesforce

`mautic.NOTICE: Doctrine\DBAL\Exception\SyntaxErrorException: An exception occurred while executing 'SELECT i.integration_entity_id, i.integration_entity, i.id, i.internal_entity_id,l. FROM integration_entity i INNER JOIN companies l ON l.id = i.internal_entity_id WHERE (i.integration = ?) AND (i.integration_entity = ?) AND (i.internal_entity = ?) AND ((i.integration_entity_id IS NOT NULL) AND (((i.last_sync_date IS NOT NULL) AND (l.date_modified > i.last_sync_date)) OR ((i.last_sync_date IS NULL) AND (l.date_modified IS NOT NULL) AND (l.date_modified > l.date_added)))) AND (l.companyname IS NOT NULL) AND (l.date_modified BETWEEN ? and ?) LIMIT 100' with params ["Salesforce", "Account", "company", "2021-09-30 01:04:06", "2021-09-30 01:24:06"]:  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'FROM integration_entity i INNER JOIN companies l ON l.id = i.internal_entity_id ' at line 1 (uncaught exception)`



#### Steps to test this PR:


1. Load up [this PR](https://mautibox.com)
2. Setup Salesforce but not setup Account/Activity
3. Check Salesforce If you have some contacts/companies to sync
4. Then you can run `php bin/console mautic:integration:fetchleads --time-interval="60 minutes" --integration=Salesforce`
5. You should see error 

`mautic.NOTICE: Doctrine\DBAL\Exception\SyntaxErrorException: An exception occurred while executing 'SELECT i.integration_entity_id, i.integration_entity, i.id, i.internal_entity_id,l. FROM integration_entity i INNER JOIN companies l ON l.id = i.internal_entity_id WHERE (i.integration = ?) AND (i.integration_entity = ?) AND (i.internal_entity = ?) AND ((i.integration_entity_id IS NOT NULL) AND (((i.last_sync_date IS NOT NULL) AND (l.date_modified > i.last_sync_date)) OR ((i.last_sync_date IS NULL) AND (l.date_modified IS NOT NULL) AND (l.date_modified > l.date_added)))) AND (l.companyname IS NOT NULL) AND (l.date_modified BETWEEN ? and ?) LIMIT 100' with params ["Salesforce", "Account", "company", "2021-09-30 01:04:06", "2021-09-30 01:24:06"]:  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'FROM integration_entity i INNER JOIN companies l ON l.id = i.internal_entity_id ' at line 1 (uncaught exception)`
